src/selenium_screenshot.py

```diff
@@ -0,0 +1,134 @@
+        browser.set_window_size(window_width, window_height)
+        browser.get(url)
+
+        return browser
+    except Exception as ex:
+        return None
+
+
+def resize_chromium_viewport(browser, height):
+    """
+        Change the viewport value of  iframe content only when the document has just one
+
+        Keyword arguments:
+        browser -- Selenium browser instance
+        height -- Browser height
+    """
+    window_height = int(height) if height is not None else 500
+    # Run javasript script to get the iframe's body height
+    new_height = browser.execute_script(
+        """
+        var iframes = document.querySelectorAll('iframe');
+        if (iframes.length === 1) {
+            var canvas = iframes[0];
+            return canvas.contentDocument.body.scrollHeight;
+        }
+        var VERTICAL_MARGIN = 16;
+        return Math.max(arguments[0], document.body.scrollHeight + VERTICAL_MARGIN);
+    """,
+        window_height,
+    )
+
+    if window_height != new_height:
+        # Append a custom command with current browser session
+        browser.command_executor._commands["send_command"] = (
+            "POST",
+            "/session/{0}/chromium/send_command".format(browser.session_id),
+        )
+        # Emulation.setDeviceMetricsOverride is a command which allows us to change the viewport size
+        params = {
+            "cmd": "Emulation.setDeviceMetricsOverride",
+            "params": {
+                "mobile": False,
+                "width": 1366,
+                "height": max(new_height, window_height),
+                "deviceScaleFactor": 1,
+                "screenOrientation": {"angle": 0, "type": "portraitPrimary"},
+            },
+        }
+        # Run custom command
+        browser.execute("send_command", params)
+
+
+def take_selenium_screenshot(url, file_name, width=1000, height=500, **kwargs):
+    app_selector_xpath = kwargs.get("app_selector_xpath", "//div[@id='root']")
+    not_load_selector_xpath = kwargs.get(
+        "not_load_selector_xpath", "//div[contains(@class, 'selenium-data-not-loaded')]"
+    )
+    # Init chromium
+    browser = _init_browser(url, width, height)
+    try:
+        # Wait until the root element is loaded
+        WebDriverWait(browser, 10000).until(
+            EC.presence_of_element_located((By.XPATH, app_selector_xpath))
+        )
+        time.sleep(1)
+
+        # Check if the element is empty
+        empty_elements = browser.find_elements_by_xpath(not_load_selector_xpath)
+        if len(empty_elements) > 0:
+            raise "Data was not loaded, selection `#root .selenium-data-not-loaded` detected."
+        time.sleep(1)
+
+        # Resize window for canvas widgets
+        resize_chromium_viewport(browser, height)
+
+        # Wait a little bit for DOM changes
+        time.sleep(2)
+
+        # Take the screenshot
+        _take_screenshot(browser, file_name)
+    except Exception as ex:
```

Comment:
Same here, isn't this a violation of the `broad-except` rule?

Reply:
Yes, it is. But I don't know and I'm not sure which errors will be throw, then I preferred to use `Exception`.